.github/workflows/pr-test.yml

```diff
@@ -185,54 +183,97 @@ jobs:
         with:
           name: tests-build-cache-${{ github.run_number }}
           path: .nx
-      - name: Build
-        run: npm run compile
-      - name: Unit tests (Full)
-        if: matrix.code-coverage
-        run: npm run test
+      - name: Compile (Delta)
+        run: npm run compile:ci:affected
       - name: Unit tests (Delta)
-        if: ${{ !matrix.code-coverage }}
-        run: npm run test:ci:changed
-      - name: Report Coverage
-        if: ${{ matrix.code-coverage && !cancelled()}}
-        uses: codecov/codecov-action@v5
-        env:
-          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+        run: npm run test:ci:affected
+      - name: Test All Versions (Delta)
+        run: npm run test-all-versions:ci:affected
+      - name: Upload Test Artifacts
+        uses: actions/upload-artifact@v4
         with:
-          verbose: true
+          name: tests-coverage-cache-${{ github.run_number }}-${{ matrix.node }}
+          include-hidden-files: true
+          if-no-files-found: error
+          retention-days: 1
+          path: |
+            !node_modules
+            **/.nyc_output/**
 
-  browser-test:
-    needs: build-and-cache
-    strategy:
-      fail-fast: false
-      matrix:
-        node: ["22"]
+  test-coverage-report:
     runs-on: ubuntu-latest
-    env:
-      NPM_CONFIG_UNSAFE_PERM: true
-      NODE_OPTIONS: --max-old-space-size=4096
+    needs: unit-and-tav-test
     steps:
       - name: Checkout
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
       - uses: actions/setup-node@v4
         with:
-          node-version: ${{ matrix.node }}
+          node-version: 18
       - name: Install
         run: npm ci
-      - name: Download Build Artifacts
+      # TODO: add the rest of versions (18.19.0 & 20.6.0)? or are these enough??
```

Comment:
if easy to do we could also use these for completeness, but just using 18 and 20 is likely more than okay 🙂